gtk/gtk-play.c

```diff
@@ -1051,6 +1051,47 @@ create_tracks_menu (GtkPlay * play, GstPlayerMediaInfo * media_info, GType type)
   return menu;
 }
 
+static gboolean
+player_set_rate_cb (GtkRange * range, GtkScrollType scroll,
+    gdouble value, GtkPlay * play)
+{
+  gst_player_set_rate (play->player, value);
+}
+
+static void
+player_speed_clicked_cb (GtkButton * button, GtkPlay * play)
+{
+  GtkWidget *scale;
+  GtkWidget *dialog;
+  GtkWidget *content;
+  GtkWidget *box;
+  gdouble value;
+
+  dialog = gtk_dialog_new ();
+  gtk_window_set_transient_for (GTK_WINDOW (dialog), GTK_WINDOW (play));
+  gtk_window_set_title (GTK_WINDOW (dialog), "Playback speed control");
+
+  content = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
+  box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 1);
+  gtk_box_pack_start (GTK_BOX (content), box, TRUE, TRUE, 0);
+
+  scale = gtk_scale_new_with_range (GTK_ORIENTATION_HORIZONTAL,
```

Comment:
I did thought about it but then inserting slider or spinbutton inside the menu itself will create usability and accessibility problems.  e.g it would interface with expected  keyboard navigation behavior when add text into text spinbox.  Those are some reason behind using dialog box. In general i think our gui controls placement needs a bit rework. Since now our feature list are growing hence it maybe good idea to just revisit on what goes where.. so far we just took a easy path on putting all the stuff inside either popup menu item or created button in control panel ;)